src/data/nat/basic.lean

```diff
@@ -1399,4 +1399,62 @@ lemma decreasing_induction_succ_left {P : ℕ → Sort*} (h : ∀n, P (n+1) →
 by { rw [subsingleton.elim mn (le_trans (le_succ m) smn), decreasing_induction_trans,
          decreasing_induction_succ'] }
 
+/-- `log hb n`, with `hb : b > 1`, is the logarithm of natural number
+`n` in base `b`. If `n` is not a power of `b`, it returns the
+largest `k:ℕ` such that `b^k ≤ n`. -/
+def log {b : ℕ} (hb : b > 1) : ℕ → ℕ
```

Comment:
For most functions like this in mathlib, we totalize them rather than taking a prop hypothesis. That is, we'd have `log : nat -> nat -> nat` that returns 0 (or whatever is convenient) on `b <= 1`.

Reply:
And in either case, all occurrences of `b > 1` should become `1 < b`; simp-normal form prefers `<` to `>`. Unfortunately the linter for this is weakened right now because of bounded quantifier syntax.